main.cpp

```diff
@@ -383,12 +449,83 @@ void Computer::saveDocument(std::string filePath)
 void Computer::playVideo(std::string videoPath)
 {
     std::string newPath { "c:/" + videoPath };
+    std::cout << "Computer::playVideo() " << newPath << std::endl;
 }
 
 
 int main()
 {
     Example::main();
+
+    Dishwasher dishWasher1;
+    Dishwasher dishWasher2;
+    Desk desk1;
+    Mp3Player mp3Player1;
+    Mp3Player mp3Player2;
+    Bicycle bicycle1;
+    CdDrive cdDrive1;
+    CdDrive::CD darkSide;
+    Monitor monitor1;
+    Monitor monitor2;
+    HardDrive hardDrive1;
+    Keyboard keyboard1;
+    Cpu cpu1;
+    Cpu cpu2;
+    Computer computer1;
+
+    dishWasher1.sprayDishes();
+    dishWasher1.spinNozzles();
+    dishWasher2.heatDishes(10);
     
+    desk1.storeThings(50);
+    desk1.holdComputer();
+    desk1.extendKeyboardTray();
+    
+    mp3Player1.loadFile("music/mysong.mp3");
+    mp3Player2.playFile();
+    mp3Player1.displayInfo();
+
+    bicycle1.supportRider();
+    bicycle1.roll();
+    bicycle1.turnLeft();
+
+    cdDrive1.playCd();
+    cdDrive1.playDVD();
+    cdDrive1.writeCdr();
+
+    darkSide.getScratched();
+    darkSide.storeData();
+    darkSide.goObsolete();
+
+    monitor1.displayImage();
+    monitor1.adjustBrightness(85);
+    monitor2.turnOff();
+
+    hardDrive1.storeData("mydocuments/word.doc");
+    hardDrive1.retrieveData("desktop/text.txt");
+    hardDrive1.goIntoPowerSaveMode();
+
+    keyboard1.outputButtonPress(23);
+    keyboard1.displayNumLock();
+    keyboard1.displayCapsLock();
+
+    cpu1.communicateWithMemory(hardDrive1);
+    cpu1.communicateWithVideoCard(monitor1);
+    cpu1.communicateWithPeripheral(keyboard1);
+
+    computer1.loadDocument("documents/mydocument.doc");
+    computer1.saveDocument("documents/myotherdocument.doc");
+    computer1.playVideo("videos/mycoolvideo.mpg");
+    
+    std::cout << "dishWasher1.startButtonWidth: " << dishWasher1.startButtonWidth << std::endl;
+    std::cout << "desk1.numberOfDrawers: " << desk1.numberOfDrawers << std::endl;
+    std::cout << "mp3Player1.onOffSwitchColor: " << mp3Player1.onOffSwitchColor << std::endl;
+    std::cout << "bicycle1.numberOfWheels: " << bicycle1.numberOfWheels << std::endl;
+    std::cout << "cdDrive1.maxSpeed: " << cdDrive1.maxSpeed << std::endl;
+    std::cout << "monitor1.depth: " << monitor1.depth << std::endl;
+    std::cout << "hardDrive1.ageInYears: " << hardDrive1.ageInYears << std::endl;
+    std::cout << "keyboard1.buttonColor: " << keyboard1.buttonColor << std::endl;
+    std::cout << "cpu1.speed: " << cpu1.speed << std::endl;
```

Comment:
Don't declare things before you use them. 
you're declaring all variables at the top, but are using them much lower down.
Declaring all variables at the top of a function is an antiquated C-style of programming. 

The modern C++ guidelines state that you should declare a variable where you use it, and limit the scope of that variable as much as possible.
That means, for example, declare the `hardDrive1` **right before** you call `hardDrive1.storeData(`
- [ ] revise

Reply:
i did not know that was antiquated, good to know.